### ISSUE TITLE
The secret is presented in clear text in apk / app bundle

### ISSUE THREAD

**Issue by evangeloszotos** [EQuimper/blog#3]
Hi,

I tested out your approach, unfortunately it just seems to be secure, though, the secret is **still presented in clear text**.

with expo SDK 40:

1. Add your key to the app.json
![image](https://user-images.githubusercontent.com/33234890/104446172-5982c900-559a-11eb-8e98-cf5a3fc2b1f3.png)

2. build your app with `expo build:android`
3. download your .apk / .aab
4. unzip your .apk / .aab
5. go to: %APKROOT%/base/assets/app.manifest 
6. open it with a text editor
![image](https://user-images.githubusercontent.com/33234890/104446472-c9914f00-559a-11eb-82af-cb15d24004f0.png)

And you can see your secret string.

Sites like https://apkpure.com/ will pull your app from the app store and list it for download in every version once you published it.

So this is to be aware, that your secret is not safe in the app.json

As anyone can obtain access to it.

Cheers

**Comment by EQuimper** [comment]
@evangeloszotos yes I know, that's why I didn't published this post. That was kind of a draft. We did see it also and wasn't happy with the result. Now we manage it either with react-native-config or we go with all the secret in server. Sorry for missed information here :(

**Comment by evangeloszotos** [comment]
@EQuimper thank you for your response.

Keeping all your secrets in the server (as you mentioned), might be the most secure approach.

Retrieving keys from the server and storing them locally using `expo-secure-store` looks like the most secure approach for local secrets to me at the moment. 

Your (draft) blog post ist actually the frist hit on google, when I search for "expo secret"
![image](https://user-images.githubusercontent.com/33234890/104468876-4121a700-55b8-11eb-88e2-1928a8cdd902.png)

It is kind of dangerous for unexperienced users. The 'draft' flag can be easily overseen (as I did).
Further, one might not know if it is a draft, because it is insecure or just because you want to modify some text.

The blog post really suggest to be a secure approach for **secrets in expo** and goolge ranks it pretty high. 

Maybe you could consider editing the blog post and mention that the approach is just a way of storing/accessing some config, but does not really meet security requirements. 

`react-native-config` (as you mentioned) is probably a better fit for config than the app.json

It might be worth it mentioning those approaches/packages in the blog post.

best regards

**Comment by EQuimper** [comment]
Thank you I will delete the post then, wow didn't know my post was there. Sorry about that
